tasks/main.yml

```diff
@@ -158,6 +158,13 @@
   tags:
     - ufw
 
+- name: Combine harden_linux_ufw_rules and harden_linux_ufw_rules_user (if defined)
+  set_fact:
+    harden_linux_ufw_rules: "{{ harden_linux_ufw_rules | union(harden_linux_ufw_rules_user|default({})) }}"
```

Comment:
You need `combine` filter here as you need to combine two hashes here and want to allow keys defined in `harden_linux_ufw_rules`  to be overridden by the same keys defined in `harden_linux_ufw_rules_user`.